### PR TITLE
Fixed reset behavior on Product form

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | We can't use button with type `reset` in product page, except if we want to reset the complete global form, this contribution is about reset the specific price rule form _only_. |
| Type? | bug fix |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | no |
| How to test? | ping @vincentbz  :- ) |

#### Important guidelines
- Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
- Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
- Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

### CODE REVIEW

**Comment by vincentbz** [comment]
Tested and ok, thanks !

File: src/PrestaShopBundle/Form/Admin/Product/ProductSpecificPrice.php [hunk 1]
```diff
@@ -186,10 +186,14 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             'label' => $this->translator->trans('Apply', array(), 'Admin.Actions'),
             'attr' => array('class' => 'btn-primary-outline js-save'),
         ))
-        ->add('cancel', 'Symfony\Component\Form\Extension\Core\Type\ResetType', array(
+        ->add('cancel', 'Symfony\Component\Form\Extension\Core\Type\ButtonType', array(
             'label' => $this->translator->trans('Cancel', array(), 'Admin.Actions'),
             'attr' => array('class' => 'btn-default-outline js-cancel'),
         ));
+        /*
+         * note: ResetType can't be used because the product page is wrapped inside a big form
```
Comment (julienbourdeau) [inline src/PrestaShopBundle/Form/Admin/Product/ProductSpecificPrice.php]: 👍

Comment (mickaelandrieu) [inline src/PrestaShopBundle/Form/Admin/Product/ProductSpecificPrice.php]: :cat2:

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 2]
```diff
@@ -610,6 +610,13 @@ var specificPrices = (function() {
     });
   }
 
+  function resetForm() {
+    /**
```
Comment (Shudrum) [inline admin-dev/themes/default/js/bundle/product/form.js]: Not the right comment type, should be `//`

Comment (mickaelandrieu) [inline admin-dev/themes/default/js/bundle/product/form.js]: This is intended. From my own experience, multi lines comment are more visible. This comment is *VERY* important, regarding the impact on the back office.

Comment (Shudrum) [inline admin-dev/themes/default/js/bundle/product/form.js]: https://github.com/airbnb/javascript#comments

Multiline comments will only be used for methods & functions descriptions. Not for this.

File: src/PrestaShopBundle/Form/Admin/Product/ProductSpecificPrice.php (hunk 1 above)
Comment (Shudrum) [inline src/PrestaShopBundle/Form/Admin/Product/ProductSpecificPrice.php]: Same here. And `note:` is a little bit redundant on a comment ^^

File: admin-dev/themes/default/js/bundle/product/form.js (hunk 2 above)
Comment (mickaelandrieu) [inline admin-dev/themes/default/js/bundle/product/form.js]: As you like sir.

File: src/PrestaShopBundle/Form/Admin/Product/ProductSpecificPrice.php (hunk 1 above)
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Form/Admin/Product/ProductSpecificPrice.php]: This is perfectly valid in PHP. More, this is a multiline comment and apply this comment 
on top of this class make the chances to be read close to 0.

I agree on the redundancy of ``note``, removed 👍

**Comment by mickaelandrieu** [comment]
@Shudrum comments adressed, thanks for review !

**Comment by mickaelandrieu** [comment]
@Shudrum rebased :)

**Comment by maximebiloe** [comment]
Thank you Mickaël!

**Pull request merged**